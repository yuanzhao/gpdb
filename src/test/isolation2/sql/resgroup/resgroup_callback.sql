-- start_ignore
DROP RESOURCE GROUP rg_callback_test;
-- end_ignore

-- process callbacks on ABORT
BEGIN;
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
DROP RESOURCE GROUP rg_callback_test;
ABORT;

-- process callbacks on COMMIT
BEGIN;
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
DROP RESOURCE GROUP rg_callback_test;
COMMIT;

-- process callbacks on ABORT after CREATE
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
BEGIN;
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
DROP RESOURCE GROUP rg_callback_test;
ABORT;

-- process callbacks on COMMIT after CREATE
BEGIN;
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
DROP RESOURCE GROUP rg_callback_test;
COMMIT;

-- process callbacks on ABORT without DROP
BEGIN;
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ABORT;

-- process callbacks on COMMIT without DROP
BEGIN;
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
COMMIT;

DROP RESOURCE GROUP rg_callback_test;
