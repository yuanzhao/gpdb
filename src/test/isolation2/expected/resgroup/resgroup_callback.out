-- start_ignore
DROP RESOURCE GROUP rg_callback_test;
ERROR:  resource group "rg_callback_test" does not exist
-- end_ignore

-- process callbacks on ABORT
BEGIN;
BEGIN
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
CREATE
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
DROP RESOURCE GROUP rg_callback_test;
DROP
ABORT;
ABORT

-- process callbacks on COMMIT
BEGIN;
BEGIN
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
CREATE
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
DROP RESOURCE GROUP rg_callback_test;
DROP
COMMIT;
COMMIT

-- process callbacks on ABORT after CREATE
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
CREATE
BEGIN;
BEGIN
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
DROP RESOURCE GROUP rg_callback_test;
DROP
ABORT;
ABORT

-- process callbacks on COMMIT after CREATE
BEGIN;
BEGIN
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
DROP RESOURCE GROUP rg_callback_test;
DROP
COMMIT;
COMMIT

-- process callbacks on ABORT without DROP
BEGIN;
BEGIN
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
CREATE
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
ABORT;
ABORT

-- process callbacks on COMMIT without DROP
BEGIN;
BEGIN
CREATE RESOURCE GROUP rg_callback_test WITH (concurrency=10, cpu_rate_limit=0.1, memory_limit=0.1);
CREATE
ALTER RESOURCE GROUP rg_callback_test SET concurrency 20;
ALTER
COMMIT;
COMMIT

DROP RESOURCE GROUP rg_callback_test;
DROP
